crates/data_components/src/s3_vectors/query_provider.rs

```diff
@@ -340,9 +352,56 @@ async fn query_vector_stream(
             )
         })?;
 
-    let num_vectors = vectors.len();
+    let num_vectors = query_vectors.len();
+
+    // Get the vector data for each output using GetVectors API.
+    let keys = query_vectors.iter().map(|v| v.key.clone()).collect();
+    let GetVectorsOutput {
+        vectors: output_vectors,
+        ..
+    } = client
+        .get_vectors(
+            GetVectorsInput::builder()
+                .set_keys(Some(keys))
+                .set_vector_bucket_name(bucket_name.clone())
+                .set_index_arn(arn.clone())
+                .set_index_name(index_name.clone())
+                .set_return_data(Some(true))
+                .build()
+                .boxed()
+                .map_err(DataFusionError::External)?,
+        )
+        .instrument(info_span!(
+            target: "task_history",
+            "s3_get_vectors",
+            bucket_name = bucket_name,
+            index_name = index_name,
+            arn = arn,
+        ))
+        .await
+        .map_err(|e| {
+            DataFusionError::External(
+                Error::S3VectorGetVectorsError {
+                    source: e.into_service_error(),
+                }
+                .into(),
+            )
+        })?;
+
+    // Put the vector data in the query_vectors
+
+    // Would be nice to zip these to avoid the nested loop and clone but don't know that they come back in the same order
```

Comment:
There could be a change between two events as well so new vector could be added, so query_vector.key == output_vector.key is required IMO